server/yamlhandler.go

```diff
@@ -58,13 +62,51 @@ func convertYAMLRequestToJSONRequest(r *http.Request) error {
 	if err != nil {
 		return err
 	}
+
 	body, err := yaml.YAMLToJSON(data)
 	if err != nil {
 		return err
 	}
+
+	valid, err := validateJsonAgainstSchema(string(body))
+	if !valid {
+		//logger.Errorf("cannot validate against json schema: %+v", err)
+		return err
+	}
+
 	r.Body = ioutil.NopCloser(bytes.NewReader(body))
 	r.ContentLength = int64(len(body))
 	r.Header.Set(httputil.HeaderContentType, httputil.MediaTypeJson)
 
 	return nil
 }
+
+func validateJsonAgainstSchema(data string) (bool, error) {
+	fileSchema, err := getJsonSchema("flow-schema.json")
+	if err != nil {
+		return false, err
+	}
+
+	loader := gojsonschema.NewReferenceLoader("file://" + fileSchema)
+	document := gojsonschema.NewStringLoader(data)
+	result, _ := gojsonschema.Validate(loader, document)
```

Comment:
We are ignoring this error here, we should handle it.

```
         path, err := fPath("flow-schema.json")
	if err != nil {
		return false, err
	}

	loader := gojsonschema.NewReferenceLoader("file://" + path)
	document := gojsonschema.NewStringLoader(data)
	result, err := gojsonschema.Validate(loader, document)
	if err != nil {
		switch err.(type) {
		case *os.PathError:
			return false, fmt.Errorf("file not found %s", path)
		default:
			return false, err
		}
	}
```

Reply:
If we follow this approach, we don't need `getJsonSchema` and we can return a nicer error from `gojsonschema.Validate`

Reply:
Fixed